trieste/models/gpflow/models.py

```diff
@@ -27,9 +27,10 @@
 from ...data import Dataset
 from ...types import TensorType
 from ...utils import DEFAULTS, jit
-from ..interfaces import FastUpdateModel, TrainableProbabilisticModel
+from ..interfaces import FastUpdateModel, TrainableProbabilisticModel, TrajectorySampler
```

Comment:
same as with `FastUpdateModel` we should probably have `ReparametrizationSamplerModel` and `TrajectorySamplerModel` interface - then we can also be more precise in each acquisition function what type of probabilistic model user needs 

(although we still need to check existence of methods at runtime as users might not use mypy)